src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1995,8 +1995,17 @@ void cache(CSharpSyntaxNode bindableRoot, BoundNode boundRoot, NullableWalker.Sn
                 if (!Compilation.NullableSemanticAnalysisEnabled) return;
 #endif
                 GuardedAddBoundTreeForStandaloneSyntax(bindableRoot, boundRoot, snapshotManager, remappedSymbols);
+
             }
 
+            static bool shouldAvoidRewrite(CSharpCompilation comp) =>
+#if DEBUG
+                !comp.NullableSemanticAnalysisEnabled;
```

Comment:
Putting this comment back in line after rebasing:

> Doesn't this disable a good chunk of testing we were trying to do in DEBUG?

Reply:
The only portion that is disabled is the actual rewrite step. I does technically add more coverage, but in practice I don't believe that's a huge gain. The rewriter is almost entirely generated and does very little validation. This codepath will still run the `DebugVerifier`, which is responsible for making sure that we actually generate the info we thing we need.

---
In reply to: [438466952](https://github.com/dotnet/roslyn/pull/42403#discussion_r438466952) [](ancestors = 438466952)